sklearn/decomposition/_pca.py

```diff
@@ -655,9 +666,10 @@ def _fit_truncated(self, X, n_components, svd_solver):
 
         # Workaround in-place variance calculation since at the time numpy
         # did not have a way to calculate variance in-place.
-        N = X.shape[0] - 1
-        X **= 2
-        total_var = xp.sum(xp.sum(X, axis=0) / N)
+        if total_var is None:
+            N = X.shape[0] - 1
+            X **= 2
+            total_var = xp.sum(xp.sum(X, axis=0) / N)
```

Comment:
```suggestion
            total_var = xp.sum(X) / N
```

Reply:
I've made this change, but the code was like this before my PR.

Maybe it was here for numeric stability? I would assume `(X**2).sum()` can get quite large. Tests don't seem to fail locally though.